include/seqan3/core/simd/view_to_simd.hpp

```diff
@@ -0,0 +1,719 @@
+     * If the target vector length is the same as the max vector length nothing will be unpacked and the
+     * function is a noop. In the other cases the corresponding parts of the source vector are extracted and
+     * upcasted to the target simd type.
+     */
+    auto unpack(max_simd_type const & row) const
+    {
+        if constexpr (chunk_size == simd_traits<max_simd_type>::length / 2)  // upcast into 2 vectors.
+        {
+            return std::array{simd::upcast<simd_t>(extract_halve<0>(row)),  // 1. halve
+                                simd::upcast<simd_t>(extract_halve<1>(row))}; // 2. halve
+        }
+        else if constexpr (chunk_size == simd_traits<max_simd_type>::length / 4) // upcast into 4 vectors.
+        {
+            return std::array{simd::upcast<simd_t>(extract_quarter<0>(row)),  // 1. quarter
+                                simd::upcast<simd_t>(extract_quarter<1>(row)),  // 2. quarter
+                                simd::upcast<simd_t>(extract_quarter<2>(row)),  // 3. quarter
+                                simd::upcast<simd_t>(extract_quarter<3>(row))}; // 4. quarter
+        }
+        else if constexpr (chunk_size == simd_traits<max_simd_type>::length / 8) // upcast into 8 vectors.
+        {
+            return std::array{simd::upcast<simd_t>(extract_eighth<0>(row)),   // 1. eighth
+                                simd::upcast<simd_t>(extract_eighth<1>(row)),   // 2. eighth
+                                simd::upcast<simd_t>(extract_eighth<2>(row)),   // 3. eighth
+                                simd::upcast<simd_t>(extract_eighth<3>(row)),   // 4. eighth
+                                simd::upcast<simd_t>(extract_eighth<4>(row)),   // 5. eighth
+                                simd::upcast<simd_t>(extract_eighth<5>(row)),   // 6. eighth
+                                simd::upcast<simd_t>(extract_eighth<6>(row)),   // 7. eighth
+                                simd::upcast<simd_t>(extract_eighth<7>(row))};  // 8. eighth
+        }
+        else
+        {
+            return std::array{simd::upcast<simd_t>(row)};
+        }
+    }
+
+    /*!\brief Unpacks the matrix of simd types and caches the respective chunk entries.
+     * \param matrix The transposed byte matrix from the efficient load procedure.
+     *
+     * \details
+     *
+     * In the efficient load procedure a quadratic byte matrix is first filled and then transposed using e
+     * efficient simd instructions. Depending on the target simd vector type this byte matrix must be unpacked
+     * and the corresponding simd vectors must be assigned to their respective position within the cached
+     * chunk array.
+     */
+    constexpr void split_into_sub_matrices(std::array<max_simd_type, simd_traits<max_simd_type>::length> matrix) const
+    {
+        auto apply_padding = [this] (simd_t const vec)
+        {
+            return (vec == simd::fill<simd_t>(static_cast<uint8_t>(~0))) ? this_view->cached_padding_vector : vec;
```

Comment:
I don't understand why this comparison is needed at all.

Reply:
Because, the paddding value is something based on the target scalar type. So it might be that the padding value is not representable in one byte. To mask the cells that need to be padded they will be set to -1 since 0 is a valid rank in the alphabets. Accordingly, we now use the padding value or the the stored value in the sime vec.

Reply:
Btw, you solution will still not work. I really don't understand the problem with static cast here. 
It won't work because of narrowing conversions.